src/dmd/console.d

```diff
@@ -97,21 +119,54 @@ struct Console
             if (GetConsoleScreenBufferInfo(h, &sbi) == 0) // get initial state of console
                 return null;
 
+            // try enabling support for ANSI escape codes
+            DWORD mode;
+            bool vtmode;
+            bool resetVtmode;
+            if (GetConsoleMode(h, &mode))
+            {
+                if (mode & ENABLE_VIRTUAL_TERMINAL_PROCESSING)
```

Comment:
This looks like a heuristic at best. Your PR description says: "combined with other build tools", which I understood to mean tools that capture output and display it later (like Ninja), or IDEs which don't run things in a console window at all. 

In which case:

- Enabling VT processing, outputting VT codes, disabling VT processing, and allowing the tool to print them will just dump unprocessed VT codes to the user.
- If the output is redirected to a file, then this check (or changing console settings) is pointless.
- Same as when output is sent to a stream, to be displayed in an IDE output pane.

Perhaps allow enabling VT sequence output using `-color=vt` ?

Reply:
I just noticed this PR does not help with the color codes with tools like Ninja, because the colors are not added when no valid Console is present (isatty fails, GetConsoleMode fails, etc.). I think the best way to deal with this is to have ``-color=vt`` or ``-color=always`` which enables the colors in all cases.